### PR TITLE
lines_of_action: surface full move history and the lose-by-disruption  path

### PR DESCRIPTION
**Pull request by bovard**
The harness was passing the framework's per-agent move_history argument straight into the prompt, which only contains the calling agent's own moves. The model therefore couldn't see the opponent's prior plies past last_move, leaving it unable to spot impending twofold-repetition draws or read the opponent's recent strategy. The prompt also covered only the simultaneous-connection case for the moving-player win, omitting the symmetric loss condition where a careless capture or moving a 'blocker' piece leaves only the opponent's group connected -- which the engine resolves as an immediate loss on the mover's turn (CheckTerminalState in lines_of_action.cc).

Fix:
- Proxy state_dict() now exposes the full game's action-string history (both players, oldest first) by replaying the deserialized engine history.
- Harness prefers state["move_history"] from the proxy over the per-agent argument and renders both players' moves in chess-style numbered-pair notation.
- Prompt gains a sentence describing the only-opponent-connected loss path so the model can defend against (or seek) it.

Regression tests cover the new prompt sentence, that the proxy history is preferred when both sources are present, and that the env emits the full history on each step.

### CODE REVIEW

File: kaggle_environments/envs/open_spiel_env/games/lines_of_action/harness.py [hunk 1]
```diff
@@ -266,14 +291,20 @@ def generate_prompt(
...
 
     my_piece = "x" if player_id == 0 else "o"
-    move_history_str = " ".join(move_history) if move_history else "None"
+    # Prefer the proxy's full-game move history (both players) over the
+    # framework's `move_history` argument (this agent's moves only) -- the
+    # latter is missing the opponent's plies, which prevents the model
+    # from detecting impending twofold-repetition draws or reading the
+    # opponent's recent strategy.
+    full_history = state.get("move_history")
+    history_to_render = full_history if full_history is not None else list(move_history)
```
Comment (kaggle-agent) [inline kaggle_environments/envs/open_spiel_env/games/lines_of_action/harness.py]: Fallback path mislabels moves if it ever fires. The framework's `move_history` argument contains only the calling agent's own moves (per the comment above), but `_format_move_history` treats `moves[0], moves[2], moves[4], ...` as Black plies and `moves[1], moves[3], ...` as White plies. If we are White and the proxy's `move_history` is ever missing, we will hand the model a list of our own moves rendered as `1. <ourMove1> <ourMove2>  2. <ourMove3> <ourMove4>...`, all labeled (via the template) as starting with Black's ply — which is factually wrong and would confuse the model about who played what. In practice the proxy is registered and always supplies `move_history`, so this is dead code today, but a safer fallback would be either (a) skipping the history block entirely when only the per-agent slice is available, or (b) labeling it explicitly as 'your prior moves only'. As written, the comment promises one thing while the fallback silently does another.

File: kaggle_environments/envs/open_spiel_env/games/lines_of_action/harness.py [hunk 2]
```diff
@@ -195,6 +200,26 @@ def _format_piece_line_counts(
...
+        return "  None"
+    pairs: list[str] = []
+    for i in range(0, len(moves), 2):
+        n = i // 2 + 1
+        black = moves[i]
+        if i + 1 < len(moves):
+            pairs.append(f"{n}. {black} {moves[i + 1]}")
+        else:
+            pairs.append(f"{n}. {black}")
+    return "  " + "  ".join(pairs)
```
Comment (kaggle-agent) [inline kaggle_environments/envs/open_spiel_env/games/lines_of_action/harness.py:218]: Minor: with the 1000-ply game cap, `_format_move_history` can emit a single ~10 KB line of `"  ".join(...)` pairs. Most models will cope, but consider `"\n".join(pairs)` (one numbered pair per line) — it scales the same in tokens, keeps the chess-style framing, and is dramatically easier for the model to scan when searching for repetitions (which is the stated motivation for surfacing the full history).

**Review by kaggle-agent — COMMENTED** [review]
**review-harness**: Prompt clarification matches LoA's suicide rule; full-history switch is well-motivated and matches what the proxy provides. One latent issue: if the proxy ever fails to populate `move_history`, the fallback feeds per-agent moves into a renderer that labels them as alternating Black/White pairs, which would silently mislabel every move.

**Review by lipovetz — APPROVED** [review]
